FiftyCans/GameScene.m

```diff
@@ -7,43 +7,172 @@
 //
 
 #import "GameScene.h"
+static const uint32_t projectileCategory     =  0x1 << 0;
+static const uint32_t monsterCategory        =  0x1 << 1;
+
+@interface GameScene() <SKPhysicsContactDelegate>
+@property (nonatomic) SKSpriteNode *player;
+@property (nonatomic) NSTimeInterval lastSpawnTimeInterval;
+@property (nonatomic) NSTimeInterval lastUpdateTimeInterval;
+@end
+
+static inline CGPoint rwAdd(CGPoint a, CGPoint b) {
+    return CGPointMake(a.x + b.x, a.y + b.y);
+}
+
+static inline CGPoint rwSub(CGPoint a, CGPoint b) {
+    return CGPointMake(a.x - b.x, a.y - b.y);
+}
+
+static inline CGPoint rwMult(CGPoint a, float b) {
+    return CGPointMake(a.x * b, a.y * b);
+}
+
+static inline float rwLength(CGPoint a) {
+    return sqrtf(a.x * a.x + a.y * a.y);
+}
+
+// Makes a vector have a length of 1
+static inline CGPoint rwNormalize(CGPoint a) {
+    float length = rwLength(a);
+    return CGPointMake(a.x / length, a.y / length);
+}
 
 @implementation GameScene
 
 -(void)didMoveToView:(SKView *)view {
-    /* Setup your scene here */
-    SKLabelNode *myLabel = [SKLabelNode labelNodeWithFontNamed:@"Chalkduster"];
-    
-    myLabel.text = @"Hello, World!";
-    myLabel.fontSize = 65;
-    myLabel.position = CGPointMake(CGRectGetMidX(self.frame),
-                                   CGRectGetMidY(self.frame));
-    
-    [self addChild:myLabel];
+
+    self.backgroundColor = [SKColor colorWithRed:0.1 green:0.1 blue:0.1 alpha:1.0];
+    self.player = [SKSpriteNode spriteNodeWithImageNamed:@"console-color"];
+    self.player.position = CGPointMake(200, 150);
+    [self addChild:self.player];
+    self.physicsWorld.gravity = CGVectorMake(0,0);
+    self.physicsWorld.contactDelegate = self;
+
+    [self addCan];
+}
+
+- (void)addCan {
+    SKSpriteNode *can = [SKSpriteNode spriteNodeWithImageNamed:@"can-tuna"];
+    can.physicsBody = [SKPhysicsBody bodyWithRectangleOfSize:can.size]; // 1
```

Comment:
Oh, yeah, all the // 1's are copy paste.  I can clean all that out in the next round.